GameEngine/Game.cs

```diff
@@ -0,0 +1,77 @@
+﻿using GameEngine.Players;
+
+namespace GameEngine
+{
+    public static class Game
+    {
+        public static GameState Start(int multiplier)
```

Comment:
❔ Why are all of these static?

I feel like `state.IsOver()` is more idiomatic and should be our default position (compared to `Game.IsOver(state)`)

Reply:
Refactored to an instance class.

I defaulted to a struct and a bucket of static functions so that the distinction between valid game operations and the things you can modify directly on GameState are entirely clear. The important thing is to keep GameState (at least conceptually) immutable. We still need actual mutability for reasonable testing, but keeping actions separate from the state should be good enough to convey that intent.

Reply:
One actual reason to make these static is to be able to reason about `GameState` outside of the context of a `Game`. As we run into those, we can kajigger things to allow reasonable / idiomatic access to those functions. I just want to be really careful about keeping state representation away from state actions. State inspections are more of a grey area.

Really, this decision seems pretty academic. We can leave it either way and change it back as necessary. For now, let's do the idiomatic instance way.

The main thrust of this PR is to make sure we have a `Successor` function somewhere, because we know for sure we'll need that later. For `Successor` to work smoothly, we needed a way to `Clone` state. From there, it seemed prudent to try to separate concerns a little more thoroughly.